### PR TITLE
Edit alert ui

### PR DESCRIPTION
**Pull request by BrandonArp**
UI for editing, creating, and deleting alerts.  It's not pretty, but it's functional.

changes include:
- refactoring the pagination component to use the compose "template" mode
- new edit/create alert page
- modifying alert list to include create, edit and delete links
- API for deleting alerts
- shiny new UUID class
- fixing graph config modal

### CODE REVIEW

File: public/html/alerts/AlertsViewModel.html [hunk 1]
```diff
@@ -18,46 +18,57 @@
...
-                        <i class="fa fa-calculator"></i>
-                        <span data-bind="text: statistic, tooltip:{ statistic, tooltipClass: 'host-tip'}, attr: {title: statistic}"></span>
-                        <br/>
-                        <i class="fa fa-clock-o"></i>
-                        <span data-bind="text: period, tooltip:{ period, tooltipClass: 'host-tip'}, attr: {title: period}"></span>
-                        <i class="fa" style="float : right;" data-bind="css: contextStyle, tooltip:{ contextTip, tooltipClass: 'host-tip'}, attr: {title: contextTip}"></i>
-                    </div>
-                </div>
+            <div class="modal-body">
+                <div>Are you sure you want to delete the alert?</div>
```
Comment (vjkoskela) [inline public/html/alerts/AlertsViewModel.html]: Are you sure _that_ you want to....

File: public/html/alerts/AlertsViewModel.html [hunk 2]
```diff
@@ -18,46 +18,57 @@
...
-                        <input class="form-control" data-bind="value: searchExpression, valueUpdate: 'keyup'" type="text" placeholder="Search">
-                        <span class="glyphicon glyphicon-search form-control-feedback"></span>
-                    </div>
-                </form>
+<div id="confirm-delete-modal" class="modal fade" tabindex="-1" role="dialog">
+    <div class="modal-dialog" role="document">
+        <div class="modal-content">
+            <div class="modal-header">
+                <button type="button" class="close" data-dismiss="modal" aria-label="Close"><span aria-hidden="true">&times;</span></button>
+                <h4 class="modal-title">Are you sure?</h4>
```
Comment (vjkoskela) [inline public/html/alerts/AlertsViewModel.html]: title => Confirmation

File: public/html/alerts/EditAlertViewModel.html [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+            <div class="col-md-12 col-sm-12 col-lg-12">
+                <form role="form" class="container-fluid form-horizontal col-md-8 col-sm-12 col-lg-6">
+                    <div class="form-group">
+                        <label for="nameInput">Name</label>
+                        <input class="form-control" data-bind="value: name" type="text" id="nameInput" placeholder="Name">
+                    </div>
+                    <div class="form-group">
+                        <label for="contextInput">Context</label>
+                        <select class="form-control" data-bind="options: contexts, optionsText: 'text', optionsValue: 'value', value: context" id="contextInput"></select>
+                        <!--<input class="form-control" data-bind="value: operator" type="text" id="operatorInput" placeholder="Operator">-->
```
Comment (vjkoskela) [inline public/html/alerts/EditAlertViewModel.html]: ?

File: public/html/alerts/EditAlertViewModel.html [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+                        <input class="form-control" data-bind="value: statistic" type="text" id="statisticInput" placeholder="Statistic">
+                    </div>
+                    <div class="form-group">
+                        <label for="periodInput">Period</label>
+                        <input class="form-control" data-bind="value: period" type="text" id="periodInput" placeholder="Period">
+                    </div>
+                    <div class="form-group">
+                        <label for="operatorInput">Operator</label>
+                        <select class="form-control" data-bind="options: operators, optionsText: 'text', optionsValue: 'value', value: operator" id="operatorInput"></select>
+                        <!--<input class="form-control" data-bind="value: operator" type="text" id="operatorInput" placeholder="Operator">-->
```
Comment (vjkoskela) [inline public/html/alerts/EditAlertViewModel.html]: ?

**Comment by vjkoskela** [comment]
Minor feedback. Can't wait to see it in action.

**Comment by BrandonArp** [comment]
Closing in favor of the upcoming MQL changes.

**Pull request closed without merging**